wavecrestweb/kenworthy-machine
# #KM-27 & #KM-33: Openings section

## Description

This PR adds responsive **Openings section** to _Careers page_. TinaCMS schema was updated to include new block for this section. I added the section's content to TinaCMS, and set up data fetching. 

The **Opening section** renders a `NoOpenings` component when there are no openings currently available. In this scenario, when the **Send Resume** button is clicked, pre-populated default email client opens in a new tab.

The `OpeningsAvailable` component will render if there are any openings available. This component renders a list of current opening. Each position card has a **Full Job Description** button that redirects the user to the external url and opens it in a new tab.

I also adjusted styling for other sections of the _Careers page_ to better match wireframes. 

## Related Tickets & Documents
Closes #KM-27 & #KM-33.

## Acceptance Criteria
### #KM-27

- [x] The 'Send Us Your Resume' section includes a visible button that, when clicked, opens the user's default email client to send an email to a specified Kenworthy Machine recruitment address.
- [x] The button is accompanied by text that sets the expectation that all resumes will be welcomed and considered for future opportunities.
- [x] The email opened by the button is pre-populated with a subject line relevant to resume submission for the recruitment team's easy identification.

### #KM-33
- [x] The 'Openings' section includes a list of current openings. 
- [x] Every opening card displays title, short summary, and a **Full Job Description** button.
- [x] The *Full Job Description** button redirects the user to the external url and opens it in a new tab.

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|   `✓`  | :sparkles: New feature     |
|   `✓`  | 🎨 Style                   |
|   `✓`  | :hammer: Refactoring       |

## Mobile & Desktop Screenshots/Recordings
### Large screens
![Screen Shot 2024-03-26 at 11 35 22 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/f9acad84-960c-4061-b541-89dc491a1930)
![Screen Shot 2024-03-26 at 11 35 55 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/c502b2dc-a4a6-43b0-a753-688ed1d8717c)
### Medium screens
![Screen Shot 2024-03-26 at 11 37 19 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/66911889-d5ff-467d-9cfd-695acc4b9dc8)
![Screen Shot 2024-03-26 at 11 36 56 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/55f89175-1b81-49be-bbd9-f163f39eb740)
### Small screens
![Screen Shot 2024-03-26 at 11 47 31 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/ec584474-d459-46fc-a819-59dbac197750)
![Screen Shot 2024-03-26 at 11 51 32 AM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/5c284681-b6e9-402e-8c3a-5e1111c3fd73)

## What I learned

It was my first time using mailto url; turned out to be pretty straightforward.

Also, it seems that no matter how much time I spend on styling, there is always something that's a little off; and I discover it after merging into main 😭.
## Testing steps

1. `git switch KM-27/33/feature-openings-section`
2. `npm run dev`
3. Open http://localhost:3000/admin#/~/careers
4. Open TinaCMS editor and try to change section's content; add a new opening
5. Delete all the current openings to render **NoOpenings section** instead of **OpeningsAvailable section**
   - try to change section's content
   - click on **Send Resume** button and check that default email client opens in a new tab
   - send an email (the email's _to:_ and _subject:_ fields will be pre-populated) and check that you received that email at the address indicated in .env
   - add a new opening to check that **OpeningsAvailable section** renders instead of **NoOpenings section**
7. Test with different screen sizes (the attached screenshots show 1728 x 1117, 834 x 1194, 430 x 932 screen sizes set up via Chrome DevTools Responsive menu)

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| `✓`  | 🙅 no documentation needed    |


## What gif best describes this PR?

  ![hard](https://media.giphy.com/media/jXp6wAeHU3Ds3UoNMG/giphy.gif?cid=790b7611s31o1kwxf0udb98kcqa05pu2re5lj9tecvpz1uie&ep=v1_gifs_search&rid=giphy.gif&ct=g)
